packages/core/src/hydration/i18n.ts

```diff
@@ -155,38 +159,113 @@ export function trySerializeI18nBlock(
   }
 
   const caseQueue: number[] = [];
-  tI18n.ast.forEach((node) => serializeI18nBlock(lView, caseQueue, context, node));
-  return caseQueue.length > 0 ? caseQueue : null;
+  const disconnectedNodes = new Set<number>();
+  const disjointNodes = new Set<number>();
+  serializeI18nBlock(lView, caseQueue, disconnectedNodes, disjointNodes, context, tI18n.ast);
```

Comment:
nit: if we create an interface (see above), we can also use the same approach here and pass this data as a single unit, for example:
```suggestion
  serializeI18nBlock(lView, i18nBlockContext, context, tI18n.ast);
```